scanners/trivy/parser/parser.js

```diff
@@ -67,30 +83,123 @@ return findings;
   
 }
 
-function getCategory(target) {
-    let category = "Image Vulnerability";
-    if (target.endsWith("package-lock.json") || target == "Node.js") {
-      category = "NPM Package Vulnerability";
-    } else if (target.endsWith("Gemfile.lock")) {
-      category = "Ruby Package Vulnerability";
-    } else if (target.endsWith("Pipfile.lock")) {
-      category = "Python Package Vulnerability";
-    } else if (target.endsWith("Cargo.lock")) {
-      category = "Rust Package Vulnerability";
-    } else if (target.endsWith("Composer.lock")) {
-      category = "PHP Package Vulnerability";
-    } else if (target.endsWith("go.sum")) {
-      category = "Go Package Vulnerability";
+function parseK8sScanResults(clusterName, scanResults) {
+  return new Promise((resolve, reject) => {
+
+    var keys = Object.keys(scanResults);
+    const expectedTopLevelAttributes = ["ClusterName", "Vulnerabilities", "Misconfigurations"];
+    const found = keys.find(key => !expectedTopLevelAttributes.includes(key));
+    if (found !== undefined) {
+      reject(new Error("Unexpected attribute '" + found + "' in scan result"));
+    }
```

Comment:
The parser shows the following error:
```
Parser failed with error:
Error: Unexpected attribute 'Resources' in scan result 
at /home/app/parser-wrapper/parser/parser.js:93:14
at new Promise (<anonymous>) 
at parseK8sScanResults (/home/app/parser-wrapper/parser/parser.js:87:10)
at parse (/home/app/parser-wrapper/parser/parser.js:20:18)                                                                                   at main (/home/app/parser-wrapper/parser-wrapper.js:127:22)                                                                                  at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

for the following results file: https://gist.github.com/Ilyesbdlala/bf015907258e7013615464e98624b172#file-trivy-results-json
I'm guessing adding `Ressources` to `expectedTopLevelAttributes` should fix it. Not sure.

Reply:
Structure of scan result file has changed in the meantime. Had to do some adjustments.